### DISCUSSION TITLE
Using Turso Embedded Replicas with PocketBase for Multi-Region Deployment

### DISCUSSION THREAD

**Discussion by gabrielvincent** [pocketbase/pocketbase#5969] [Q&A] [answered]
I'm working on enabling PocketBase to alternatively use libsql and Turso as the database driver, specifically leveraging [Turso embedded replicas](https://docs.turso.tech/features/embedded-replicas/introduction). Here's my current progress and questions:

### Current Setup
- Successfully configured PocketBase to use a remote Turso database
- Implemented synchronization between the remote database and local pb_data/data.db
- Multiple PocketBase instances across different regions replicate data efficiently using libsql + Turso

### Question
Before proceeding further, I need to better understand the purpose of PocketBase's auxiliary.db. Currently, I've kept the implementation simple by not creating a remote database for it. Understanding its role will help me determine whether it needs to be replicated across instances via Turso as well.
Would anyone be able to explain the purpose and importance of auxiliary.db in PocketBase's architecture?

**Comment by ganigeorgiev** [comment, marked as answer]
The `auxiliary.db` is currently used only for storing logs (this include request logs and manual log writes via [`app.Logger()`](https://pocketbase.io/docs/go-logging/)). It could be treated as "ephemeral"/temp database because even if it gets deleted it will be recreated on the next app start.

Regarding _"Multiple PocketBase instances across different regions replicate data..."_ - note that PocketBase is not designed to work out of the box in a multi-server environment because there is no builtin synchronization mechanism for the [various app event hooks](https://pocketbase.io/docs/go-event-hooks/). For example, if a record is created only the instance that received the request will trigger the [`OnRecordCreate` hook](https://pocketbase.io/docs/go-event-hooks/#onrecordcreate) (_the same apply for the realtime events because they are implemented on top of the hooks_). 
It is possible that your use case may not need be affected by it but it is something to keep in mind.

**Reply by gabrielvincent** [reply]
I see how my setup wouldn't work with realtime events. I don't currently need that feature, but if I ever come to a point where I need it, it would be possible to use Pocketbase as framework and use Redis to achieve realtime events.
